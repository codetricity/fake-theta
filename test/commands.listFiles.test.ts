import { request, toMatchMissingParameterError } from './helper';

describe('POST /osc/commands/execute listFiles', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.listFiles',
        parameters: { fileType: 'image', entryCount: 3, maxThumbSize: 640 },
      })
      .expect(200);
    expect(res.body.results.totalEntries).toEqual(10);
    res.body.results.entries.map((entry: unknown) => {
      expect(entry).toMatchSnapshot({
        dateTime: expect.stringMatching(
          /^[0-9]{4}(:[0-9]{2}){2} [0-9]{2}(:[0-9]{2}){2}/,
        ),
        _favorite: expect.any(Boolean),
        fileUrl: expect.stringMatching(/^(http|https):\/\/.+\/.+\/.+\/.+\.JPG/),
        isProcessed: expect.any(Boolean),
        name: expect.any(String),
        previewUrl: expect.any(String),
        size: expect.any(Number),
      });
    });
  });

  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.listFiles',
        parameters: { fileType: 'image', entryCount: 3, maxThumbSize: 0 },
      })
      .expect(200);
    expect(res.body.results.totalEntries).toEqual(10);
    res.body.results.entries.map((entry: unknown) => {
      expect(entry).toMatchSnapshot({
        dateTime: expect.stringMatching(
          /^[0-9]{4}(:[0-9]{2}){2} [0-9]{2}(:[0-9]{2}){2}/,
        ),
        _favorite: expect.any(Boolean),
        fileUrl: expect.any(String),
        isProcessed: expect.any(Boolean),
        name: expect.any(String),
        previewUrl: expect.any(String),
        size: expect.any(Number),
      });
    });
  });

  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.listFiles',
        parameters: { fileType: 'video', entryCount: 3, maxThumbSize: 640 },
      })
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera.listFiles', parameters: {} })
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('THETA X POST /osc/commands/execute listFiles', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.listFiles',
        parameters: { fileType: 'image', entryCount: 3, maxThumbSize: 640 },
      })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body.results.totalEntries).toEqual(10);
    res.body.results.entries.map((entry: unknown) => {
      expect(entry).toMatchSnapshot({
        dateTime: expect.stringMatching(
          /^[0-9]{4}(:[0-9]{2}){2} [0-9]{2}(:[0-9]{2}){2}/,
        ),
        _favorite: expect.any(Boolean),
        fileUrl: expect.any(String),
        isProcessed: expect.any(Boolean),
        name: expect.any(String),
        previewUrl: expect.any(String),
        size: expect.any(Number),
      });
    });
  });

  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.listFiles',
        parameters: { fileType: 'image', entryCount: 3, maxThumbSize: 0 },
      })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body.results.totalEntries).toEqual(10);
    res.body.results.entries.map((entry: unknown) => {
      expect(entry).toMatchSnapshot({
        dateTime: expect.stringMatching(
          /^[0-9]{4}(:[0-9]{2}){2} [0-9]{2}(:[0-9]{2}){2}/,
        ),
        _favorite: expect.any(Boolean),
        fileUrl: expect.any(String),
        isProcessed: expect.any(Boolean),
        name: expect.any(String),
        previewUrl: expect.any(String),
        size: expect.any(Number),
      });
    });
  });

  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.listFiles',
        parameters: { fileType: 'video', entryCount: 3, maxThumbSize: 640 },
      })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera.listFiles', parameters: {} })
      .set('emulating-theta-model', 'x')
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('THETA Z1 POST /osc/commands/execute listFiles', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.listFiles',
        parameters: { fileType: 'image', entryCount: 3, maxThumbSize: 640 },
      })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body.results.totalEntries).toEqual(10);
    res.body.results.entries.map((entry: unknown) => {
      expect(entry).toMatchSnapshot({
        dateTimeZone: expect.stringMatching(
          /^[0-9]{4}(:[0-9]{2}){2} [0-9]{2}(:[0-9]{2}){2}[+-][0-9]{2}:[0-9]{2}/,
        ),
        fileUrl: expect.stringMatching(
          /^(http|https):\/\/.+\/.+\/.+\/.+\/.+\.JPG/,
        ),
        height: expect.any(Number),
        isProcessed: expect.any(Boolean),
        name: expect.any(String),
        previewUrl: expect.any(String),
        _projectionType: expect.stringMatching(
          /^(Equirectangular|Dual-Fisheye)/,
        ),
        size: expect.any(Number),
        _thumbSize: expect.any(Number),
        thumbnail: expect.any(String),
        width: expect.any(Number),
      });
    });
  });

  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.listFiles',
        parameters: { fileType: 'image', entryCount: 3, maxThumbSize: 0 },
      })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body.results.totalEntries).toEqual(10);
    res.body.results.entries.map((entry: unknown) => {
      expect(entry).toMatchSnapshot({
        dateTimeZone: expect.stringMatching(
          /^[0-9]{4}(:[0-9]{2}){2} [0-9]{2}(:[0-9]{2}){2}[+-][0-9]{2}:[0-9]{2}/,
        ),
        fileUrl: expect.stringMatching(
          /^(http|https):\/\/.+\/.+\/.+\/.+\/.+\.JPG/,
        ),
        height: expect.any(Number),
        isProcessed: expect.any(Boolean),
        name: expect.any(String),
        previewUrl: expect.any(String),
        _projectionType: expect.stringMatching(
          /^(Equirectangular|Dual-Fisheye)/,
        ),
        size: expect.any(Number),
        _thumbSize: expect.any(Number),
        width: expect.any(Number),
      });
    });
  });

  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.listFiles',
        parameters: { fileType: 'video', entryCount: 3, maxThumbSize: 640 },
      })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera.listFiles', parameters: {} })
      .set('emulating-theta-model', 'z1')
      .expect(400);
    toMatchMissingParameterError(res);
  });
});
