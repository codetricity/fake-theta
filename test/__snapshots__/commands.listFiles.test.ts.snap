// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/commands/execute listFiles should respond with results 1`] = `
{
  "_favorite": Any<Boolean>,
  "dateTime": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}/,
  "fileUrl": StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
}
`;

exports[`POST /osc/commands/execute listFiles should respond with results 2`] = `
{
  "_favorite": Any<Boolean>,
  "dateTime": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}/,
  "fileUrl": StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
}
`;

exports[`POST /osc/commands/execute listFiles should respond with results 3`] = `
{
  "_favorite": Any<Boolean>,
  "dateTime": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}/,
  "fileUrl": StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
}
`;

exports[`POST /osc/commands/execute listFiles should respond with results 4`] = `
{
  "_favorite": Any<Boolean>,
  "dateTime": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}/,
  "fileUrl": Any<String>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
}
`;

exports[`POST /osc/commands/execute listFiles should respond with results 5`] = `
{
  "_favorite": Any<Boolean>,
  "dateTime": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}/,
  "fileUrl": Any<String>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
}
`;

exports[`POST /osc/commands/execute listFiles should respond with results 6`] = `
{
  "_favorite": Any<Boolean>,
  "dateTime": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}/,
  "fileUrl": Any<String>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
}
`;

exports[`POST /osc/commands/execute listFiles should respond with results 7`] = `
{
  "name": "camera.listFiles",
  "results": {
    "entries": [],
    "totalEntries": 0,
  },
  "state": "done",
}
`;

exports[`THETA X POST /osc/commands/execute listFiles should respond with results 1`] = `
{
  "_favorite": Any<Boolean>,
  "dateTime": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}/,
  "fileUrl": Any<String>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
}
`;

exports[`THETA X POST /osc/commands/execute listFiles should respond with results 2`] = `
{
  "_favorite": Any<Boolean>,
  "dateTime": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}/,
  "fileUrl": Any<String>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
}
`;

exports[`THETA X POST /osc/commands/execute listFiles should respond with results 3`] = `
{
  "_favorite": Any<Boolean>,
  "dateTime": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}/,
  "fileUrl": Any<String>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
}
`;

exports[`THETA X POST /osc/commands/execute listFiles should respond with results 4`] = `
{
  "_favorite": Any<Boolean>,
  "dateTime": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}/,
  "fileUrl": Any<String>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
}
`;

exports[`THETA X POST /osc/commands/execute listFiles should respond with results 5`] = `
{
  "_favorite": Any<Boolean>,
  "dateTime": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}/,
  "fileUrl": Any<String>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
}
`;

exports[`THETA X POST /osc/commands/execute listFiles should respond with results 6`] = `
{
  "_favorite": Any<Boolean>,
  "dateTime": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}/,
  "fileUrl": Any<String>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
}
`;

exports[`THETA X POST /osc/commands/execute listFiles should respond with results 7`] = `
{
  "name": "camera.listFiles",
  "results": {
    "entries": [],
    "totalEntries": 0,
  },
  "state": "done",
}
`;

exports[`THETA Z1 POST /osc/commands/execute listFiles should respond with results 1`] = `
{
  "_projectionType": StringMatching /\\^\\(Equirectangular\\|Dual-Fisheye\\)/,
  "_thumbSize": Any<Number>,
  "dateTimeZone": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}\\[\\+-\\]\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "fileUrl": StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
  "height": Any<Number>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
  "thumbnail": Any<String>,
  "width": Any<Number>,
}
`;

exports[`THETA Z1 POST /osc/commands/execute listFiles should respond with results 2`] = `
{
  "_projectionType": StringMatching /\\^\\(Equirectangular\\|Dual-Fisheye\\)/,
  "_thumbSize": Any<Number>,
  "dateTimeZone": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}\\[\\+-\\]\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "fileUrl": StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
  "height": Any<Number>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
  "thumbnail": Any<String>,
  "width": Any<Number>,
}
`;

exports[`THETA Z1 POST /osc/commands/execute listFiles should respond with results 3`] = `
{
  "_projectionType": StringMatching /\\^\\(Equirectangular\\|Dual-Fisheye\\)/,
  "_thumbSize": Any<Number>,
  "dateTimeZone": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}\\[\\+-\\]\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "fileUrl": StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
  "height": Any<Number>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
  "thumbnail": Any<String>,
  "width": Any<Number>,
}
`;

exports[`THETA Z1 POST /osc/commands/execute listFiles should respond with results 4`] = `
{
  "_projectionType": StringMatching /\\^\\(Equirectangular\\|Dual-Fisheye\\)/,
  "_thumbSize": Any<Number>,
  "dateTimeZone": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}\\[\\+-\\]\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "fileUrl": StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
  "height": Any<Number>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
  "width": Any<Number>,
}
`;

exports[`THETA Z1 POST /osc/commands/execute listFiles should respond with results 5`] = `
{
  "_projectionType": StringMatching /\\^\\(Equirectangular\\|Dual-Fisheye\\)/,
  "_thumbSize": Any<Number>,
  "dateTimeZone": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}\\[\\+-\\]\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "fileUrl": StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
  "height": Any<Number>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
  "width": Any<Number>,
}
`;

exports[`THETA Z1 POST /osc/commands/execute listFiles should respond with results 6`] = `
{
  "_projectionType": StringMatching /\\^\\(Equirectangular\\|Dual-Fisheye\\)/,
  "_thumbSize": Any<Number>,
  "dateTimeZone": StringMatching /\\^\\[0-9\\]\\{4\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\} \\[0-9\\]\\{2\\}\\(:\\[0-9\\]\\{2\\}\\)\\{2\\}\\[\\+-\\]\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "fileUrl": StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
  "height": Any<Number>,
  "isProcessed": Any<Boolean>,
  "name": Any<String>,
  "previewUrl": Any<String>,
  "size": Any<Number>,
  "width": Any<Number>,
}
`;

exports[`THETA Z1 POST /osc/commands/execute listFiles should respond with results 7`] = `
{
  "name": "camera.listFiles",
  "results": {
    "entries": [],
    "totalEntries": 0,
  },
  "state": "done",
}
`;
